src/test/kfs_rest_test.cpp

```diff
@@ -245,21 +247,185 @@ static void testInference(int headerLength, std::string& request_body, std::uniq
 
     auto output = doc["outputs"].GetArray()[0].GetObject()["data"].GetArray();
     ASSERT_EQ(output.Size(), 10);
-
+    auto datatype = doc["outputs"].GetArray()[0].GetObject()["datatype"].GetString();
     for (auto& data : output) {
-        ASSERT_EQ(data.GetFloat(), 2);
+        if (strcmp(datatype, "BOOL") == 0) {
+            ASSERT_EQ(data.GetBool(), true);
+        } else {
+            ASSERT_EQ(data.GetFloat(), 2);
+        }
     }
 }
 
+static void testInferenceNegative(int headerLength, std::string& request_body, std::unique_ptr<HttpRestApiHandler>& handler, ovms::Status processorStatus) {
+    std::string request = "/v2/models/mediapipeAdd/versions/1/infer";
+
+    std::vector<std::pair<std::string, std::string>> headers;
+    std::pair<std::string, std::string> binaryInputsHeader{"Inference-Header-Content-Length", std::to_string(headerLength)};
+    headers.emplace_back(binaryInputsHeader);
+
+    ovms::HttpRequestComponents comp;
+
+    ASSERT_EQ(handler->parseRequestComponents(comp, "POST", request, headers), ovms::StatusCode::OK);
+
+    std::string response;
+    ovms::HttpResponseComponents responseComponents;
+    ASSERT_EQ(handler->dispatchToProcessor(request_body, &response, comp, responseComponents), processorStatus);
+}
+
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wnarrowing"
 
-TEST_F(HttpRestApiHandlerWithMediapipeForkTest, inferRequestFP32) {
+TEST_F(HttpRestApiHandlerWithMediapipe, inferRequestFP64) {
```

Comment:
There are a lot of very similar tests here. Maybe we could parametrize some of them? The only differences I see are in datatype and status.